backend/src/org/commcare/suite/model/Detail.java

```diff
@@ -438,4 +442,12 @@ public void populateEvaluationContextVariables(EvaluationContext ec) {
             ec.setVariable(key, XPathFuncExpr.unpack(variables.get(key).eval(ec)));
         }
     }
+
+    public boolean evaluateFocusFunction(EvaluationContext ec) {
+        if (focusFunction == null) {
+            return false;
+        }
+        Object value = XPathFuncExpr.unpack(focusFunction.eval(ec));
+        return XPathFuncExpr.toBoolean(value).booleanValue();
```

Comment:
@ctsims `toBoolean` will throw an `XPathTypeMismatchException` if `value` is not convertible to a boolean, so I think that should be sufficient. does that seem right?